src/Features/LanguageServer/Microsoft.CommonLanguageServerProtocol.Framework/RequestExecutionQueue.cs

```diff
@@ -175,9 +191,24 @@ private async Task ProcessQueueAsync()
                 try
                 {
                     var (work, activityId, cancellationToken) = queueItem;
+                    CancellationTokenSource? cancellationTokenSource = null;
                     lspServices = work.LspServices;
 
-                    var cancellationTokenSource = CancellationTokenSource.CreateLinkedTokenSource(CancellationToken, cancellationToken);
+                    if (CancelInProgressWorkUponMutatingRequest)
+                    {
+                        // Verify this queueitem hasn't already been cancelled before creating a linked
+                        // CancellationTokenSource based on it.
+                        if (cancellationToken.IsCancellationRequested)
+                        {
+                            cancellationToken.ThrowIfCancellationRequested();
```

Comment:
think we could just call `cancellationToken.ThrowIfCancellationRequested();` without the if

Reply:
Yeah, Cyrus also mentioned that. I added it as I had some code I was using for debugging in that condition. Went ahead and removed.